homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,633 @@
+                "Specific %s response: get_State()=%s",
+                SWITCH_TYPE_DEVICEPROFILE,
+                status,
+            )
+            if status == "never":
+                self._attr_is_on = False
+                self._is_available = True
+            elif status == "unlimited":
+                self._attr_is_on = True
+                self._is_available = True
+            else:
+                self._is_available = False
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.error("Could not get %s state", self.name, exc_info=True)
+            self._is_available = False
+
+    async def _async_switch_on_off_executor(self, turn_on: bool) -> None:
+        """Handle profile switch."""
+        state = "unlimited" if turn_on else "never"
+        await self.hass.async_add_executor_job(
+            self.fritzbox_tools.fritz_profiles[self.profile].set_state, state
+        )
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        return False
+
+
+class FritzBoxWifiSwitch(FritzBoxBaseSwitch, SwitchEntity):
+    """Defines a FRITZ!Box Tools Wifi switch."""
+
+    def __init__(
+        self,
+        fritzbox_tools: FritzBoxTools,
+        device_friendly_name: str,
+        network_num: int,
+        network_name: str,
+    ) -> None:
+        """Init Fritz Wifi switch."""
+        self._fritzbox_tools: FritzBoxTools = fritzbox_tools
+
+        self._attributes = {}
+        self._network_num = network_num
+
+        switch_info = SwitchInfo(
+            description=f"Wi-Fi {network_name}",
+            friendly_name=device_friendly_name,
+            icon="mdi:wifi",
+            type=SWITCH_TYPE_WIFINETWORK,
+            callback_update=self._async_fetch_update,
+            callback_switch=self._async_switch_on_off_executor,
+        )
+        super().__init__(self._fritzbox_tools, switch_info)
+
+    async def _async_fetch_update(self) -> None:
+        """Fetch updates."""
+
+        try:
+            wifi_info = await async_service_call_action(
+                self.fritzbox_tools,
+                "WLANConfiguration",
+                str(self._network_num),
+                "GetInfo",
+            )
+            _LOGGER.debug(
+                "Specific %s response: GetInfo=%s", SWITCH_TYPE_WIFINETWORK, wifi_info
+            )
+
+            if wifi_info is None:
+                self._is_available = False
+                return
+
+            self._attr_is_on = wifi_info["NewEnable"] is True
+            self._is_available = True
+
+            std = wifi_info["NewStandard"]
+            self._attributes["standard"] = std if std else None
+            self._attributes["BSSID"] = wifi_info["NewBSSID"]
+            self._attributes["mac_address_control"] = wifi_info[
+                "NewMACAddressControlEnabled"
+            ]
+
+        except FritzConnectionException:
```

Comment:
This code is unreachable since `async_service_call_action` traps `FritzConnectionException` and `Exception`

Reply:
Indeed, leftover of previous call